### PR TITLE
feat: Hook up frontend with backend projects

### LINKED ISSUE #208: Display projects from backend

### PR DESCRIPTION
**Pull request by david-code**
# Description

Again, pretty straightforward. Relies on https://github.com/techmatters/terraso-client-shared/pull/37 , so will have to update package.json to match that first.

### Related Issues
Fixes #208

### CODE REVIEW

File: dev-client/src/components/common/IconChip.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {Badge, HStack, Text} from 'native-base';
+import {Icon} from './Icons';
+
+type Props = {
+  iconName: string;
+  label: any;
```
Comment (shrouxm) [inline dev-client/src/components/common/IconChip.tsx:6]: nit: 
```suggestion
  label: string;
```
?

File: dev-client/src/components/home/BottomSheet.tsx [hunk 2]
```diff
@@ -15,9 +15,7 @@ type SiteListSiteProps = {
 const SiteListSite = ({site, showSiteOnMap}: SiteListSiteProps) => {
   const {t} = useTranslation();
   const project = useSelector(state =>
-    site.projectId === undefined
-      ? undefined
-      : state.project.sites[site.projectId],
+    site.projectId === undefined ? undefined : state.site.sites[site.projectId],
```
Comment (shrouxm) [inline dev-client/src/components/home/BottomSheet.tsx:18]: nit:
```suggestion
    site.projectId === undefined ? undefined : state.project.projects[site.projectId],
```
this was my bad, i fixed a silly mistake in https://github.com/techmatters/terraso-client-shared/pull/29 and didn't change the downstream weirdness created by that mistake. i think  i also fixed this in the screen refactor i just merged so it should be good if you rebase

File: dev-client/src/components/home/SiteMap.tsx [hunk 3]
```diff
@@ -42,9 +42,7 @@ type SiteCalloutProps = {
 const SiteCallout = ({site, closeCallout}: SiteCalloutProps) => {
   const {t} = useTranslation();
   const project = useSelector(state =>
-    site.projectId === undefined
-      ? undefined
-      : state.project.sites[site.projectId],
+    site.projectId === undefined ? undefined : state.site.sites[site.projectId],
```
Comment (shrouxm) [inline dev-client/src/components/home/SiteMap.tsx:45]: nit: same as prev comment
```suggestion
    site.projectId === undefined ? undefined : state.project.projects[site.projectId],
```

File: dev-client/src/components/projects/CreateProjectView/InnerForm.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import {HStack, Input, Text, VStack} from 'native-base';
```
Comment (shrouxm) [inline dev-client/src/components/projects/CreateProjectView/InnerForm.tsx:1]: nit: this file looks like code from a file that got deleted from https://github.com/techmatters/terraso-mobile-client/pull/207

File: dev-client/src/components/projects/CreateProjectView/validation.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import * as yup from 'yup';
```
Comment (shrouxm) [inline dev-client/src/components/projects/CreateProjectView/validation.ts:1]: nit: same as prev comment, i think this is code from https://github.com/techmatters/terraso-mobile-client/pull/207 that shouldn't be here

File: dev-client/src/components/projects/ProjectPreviewCard.tsx [hunk 6]
```diff
@@ -1,56 +1,49 @@
...
+  const navigation = useNavigation();
+  const dispatch = useDispatch();
 
-  const goToProject = useCallback(() => {
-    const p = fetchProject(project.id);
-    return navigation.navigate(ScreenRoutes.PROJECT_VIEW, {project: p});
-  }, [project, navigation]);
+  const goToProject = useCallback(async () => {
+    await dispatch(fetchProject(project.id));
+    return navigation.navigate(ScreenRoutes.PROJECT_VIEW);
```
Comment (shrouxm) [inline dev-client/src/components/projects/ProjectPreviewCard.tsx:24]: question (for a future PR): how should we handle this in general? this ensures the project data exists before navigating to the screen, but has the downside that we have to wait for a backend request to finish responding to the button press even if we already have the project data

**Review by shrouxm — APPROVED** [review]
the screen refactor is going to wreck this PR with conflicts i'm sorry :upside_down_face: otherwise looks good!

File: dev-client/src/components/projects/CreateProjectView/InnerForm.tsx (hunk 4 above)
Comment (david-code) [reply dev-client/src/components/projects/CreateProjectView/InnerForm.tsx:1]: Yes, I always miss this stuff :disappointed:

File: dev-client/src/components/projects/ProjectPreviewCard.tsx (hunk 6 above)
Comment (david-code) [reply dev-client/src/components/projects/ProjectPreviewCard.tsx:24]: Yeah, that's a great point. I've been thinking a bit about how we should structure the global state. I think it makes sense to take another look at this when we starting adding offline stuff. I made another issue and tentatively linked it to the offline user story. I think we can take another look whenever the moment seems right to us.

Comment (david-code) [reply dev-client/src/components/projects/ProjectPreviewCard.tsx:24]: In the meantime, I think you're right and we should use a selector to check if this project data has already been updated.

**Pull request merged**